### ISSUE TITLE
Previous layer's end region = layer's start region

### ISSUE THREAD

**Issue by Timmmmaaahh** [Ultimaker/Cura#14421]
### Is your feature request related to a problem?

Hello world!

First timer here so please be gentile :)
I have a little idea to eliminate the last travel to the nozzle's original position. As demonstrated in the gif below, after completing a layer in 3 different regions, the nozzle returns to the first region to prepare for printing the next layer. In my humble opinion this last travel is obsolete.

![Ultimaker-Cura_Suggestion](https://user-images.githubusercontent.com/19148757/214517340-ebc91114-d029-4aa7-98c5-8cdc31ae6633.gif)

### Describe the solution you'd like

It would cause less stringing and print a bit faster if the nozzle would just remain at the last region to start the new layer. Thus printing this example in the following order:

1. Layer 1: region 1 - region 2 - region 3
2. Layer 2: region 3 - region 1 - region 2
3. Layer 3: region 2 - region 3 - region 1
4. Layer 4: region 1 - region 2 - region 3
5. etc.

In other words, the last 'island' of a layer should be the first to start the next layer with.

![image](https://user-images.githubusercontent.com/19148757/214518686-726ba438-dacf-40a5-af6b-d66e13823ec8.png)

### Describe alternatives you've considered

The current way of course works just fine, this would merely be an optimization. Perhaps this would cause an increased chance of other issues due to the continuously shifting order of the starting position. I'm fairly new to 3D printing.

### Affected users and/or printers

If this wouldn't generate other issues then I'd say it would be helpful to all users and all printers.

### Additional information & file uploads

Using Cura version 5.2.1 on Windows 11 (22H2).
